### PR TITLE
Shots balancement in few shot classifiers

### PR DESCRIPTION
**Pull request by MiguelLopezCampos**
The changes in a general view are:

- New selector of samples created. It inherits from langchain example selector and its function is to feed the LLM with examples in a way that these examples have a class balancement.
- Changes that have been made to integrate this new example selection into the promptmeteo framework.
- Changes to arise errors when incongruent selector-task combination occur. This selection can only be chosen when the model is a DocumentClassifier.
- Changes in unit tests to adapt them to the new funcitonalities

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,6 +7,7 @@ __pycache__/
 *.so
 
 # Distribution / packaging
+new_env/*
```
Comment (ehooo) [inline .gitignore]: This is not standard naming, please remove it

File: promptmeteo/base.py [hunk 2]
```diff
@@ -62,6 +63,7 @@ def __init__(
         prompt_detail: Optional[str] = None,
         selector_k: int = 10,
         selector_algorithm: str = "relevance",
+        selector_k_per_class: Optional[int] = 2,
```
Comment (ehooo) [inline promptmeteo/base.py]: I'm not sure that this is the better way to do this, but i see that you need it to call `build_selector_by_load`.

Let's go with that change, but we must think in a better way to extend the base class

File: promptmeteo/selector/base.py [hunk 3]
```diff
@@ -112,10 +119,14 @@ def load_example_selector(
             self._embeddings,
         )
 
-        self._selector = self.selector(
-            vectorstore=vectorstore,
-            k=self._selector_k,
-        )
+        if self.selector == BalancedSemanticSamplesSelector:
```
Comment (ehooo) [inline promptmeteo/selector/base.py]: I think that is better call always with kwargs

**Review by ehooo — CHANGES_REQUESTED** [review]
Thanks for contribute,
Please change only the .gitignore file and the point with kwargs.

File: promptmeteo/base.py (hunk 2 above)
Comment (MiguelLopezCampos) [reply promptmeteo/base.py]: Hi! Do you mean to have 'selector_k_per_class' as object variable?

File: promptmeteo/selector/base.py (hunk 3 above)
Comment (MiguelLopezCampos) [reply promptmeteo/selector/base.py]: Hi! In this case, the selectors belong to Langchain framework and the class method from_examples does not acept kwargs arguments. How could I address this then?

File: promptmeteo/base.py (hunk 2 above)
Comment (MiguelLopezCampos) [reply promptmeteo/base.py]: This is solved :)

**Pull request merged**